### ISSUE TITLE
Clear Browsing Data dialog's search engine notice text for Brave search needs updating.

### ISSUE THREAD

**Issue by mkarolin** [brave/brave-browser#30420]
STR:
1. Start Brave
2. Navigate to brave://settings/clearBrowserData

Observe notice:

<image src="https://github.com/brave/brave-browser/assets/41635752/2c23cd2b-f983-4823-8140-1f4147163b9a" width="50%" height="50%"/>

<br/>
<br/>

The wording `See their instructions` is awkward since `their` in this case is `our`.

**Comment by rebron** [comment]
cc: @rmcfadden3

**Comment by rmcfadden3** [comment]
Yeah, @mkarolin and @rebron, agreed that this is awkward. One option would just be to wordsmith and make this message universal:

- Your search engine is [Brave]. See your search engine's settings for instructions on how to delete search history (if applicable).

Another option (which may be too complex) is to make this message dynamic so it's slightly different depending on the search engine. For example, Brave's could look like:

- Your search engine is Brave. This engine does not track search history, so there's nothing to delete. More at https://search.brave.com/help/privacy-policy.

Thoughts?

**Comment by ShivanKaul** [comment]
I think it would be best to just remove this text if the search engine is set to Brave. As mentioned, it's confusing.

**Comment by kjozwiak** [comment]
The above requires `1.89.128` or higher for `1.89.x` verification(s) 👍

**Comment by LaurenWags** [comment]
Verified with

```
Brave | 1.89.128 Chromium: 147.0.7727.25 (Official Build) beta (arm64)
-- | --
Revision | 39096907d3237d427408388a62dbec5da257544d
OS | macOS Version 26.3 (Build 25D125)
```

<details>
<summary>Case 1 - US Region - Brave default SE - PASSED</summary>

Note, a region where Brave is default SE (Canada, France, etc) could also be used in place of US

1. Launch clean profile with 1.89.x
2. Navigate to brave://setting/search
3. Confirm Brave is default SE
4. Navigate to brave://settings/clearBrowserData
5. Confirm no messaging is displayed for "Your search engine is..." on either "Delete browsing data" or "Delete data on exit" tabs

Example | Example
--- | ---
<img width="561" height="729" alt="Image" src="https://github.com/user-attachments/assets/d2feaffe-499b-4787-88b2-f3e41feb6ef4" /> | <img width="581" height="613" alt="Image" src="https://github.com/user-attachments/assets/57230c01-b00b-4c0b-997a-ac7e31f49dc6" />

</details>

<details>
<summary>Case 2 - US Region - changed region specific default SE - PASSED</summary>

1. Continue from above test case
2. Enable Rewards
3. Navigate to brave://setting/search
4. Change default SE to something other than Brave

Example | Example
--- | ---
<img width="588" height="567" alt="Image" src="https://github.com/user-attachments/assets/08c71444-bed0-4930-b215-357c34664c2e" /> | <img width="1383" height="352" alt="Image" src="https://github.com/user-attachments/assets/965860b0-0725-4d69-95df-347dd4137747" />

5. Navigate to brave://settings/clearBrowserData
6. Confirm no messaging is displayed for "Your search engine is..." on either "Delete browsing data" or "Delete data on exit" tabs

Example | Example
--- | ---
<img width="556" height="731" alt="Image" src="https://github.com/user-attachments/assets/7296b307-a915-4ae3-b4cf-71e1be840b86" /> | <img width="574" height="613" alt="Image" src="https://github.com/user-attachments/assets/e41806d7-5087-474c-bd5d-e87b4d8c3095" />

</details>

<details>
<summary>Case 3 - non-US Region - region specific default SE - PASSED</summary>

Pre-req: have your OS set to a region where Brave is not the default SE. Examples that could be used:
- Finland, Norway, etc (default SE Google)
- Russia (default SE Yandex)
- Japan (default SE Yahoo! Japan)

I used Norway/Google for below:

<img width="723" height="283" alt="Image" src="https://github.com/user-attachments/assets/37b549d2-34cb-434e-89da-87e63a3e0a6a" />

1. Launch clean profile with 1.89.x
2. Navigate to brave://setting/search
3. Confirm Google is default SE

<img width="1188" height="332" alt="Image" src="https://github.com/user-attachments/assets/d46e52c6-22d0-4ea7-bb19-282f4c4c0cfa" />

4. Navigate to brave://settings/clearBrowserData
5. Confirm no messaging is displayed for "Your search engine is..." on either "Delete browsing data" or "Delete data on exit" tabs

Example | Example
--- | ---
<img width="568" height="741" alt="Image" src="https://github.com/user-attachments/assets/f40f9f53-9306-4e07-9e33-b72822b91d8d" /> | <img width="558" height="599" alt="Image" src="https://github.com/user-attachments/assets/0ab05703-5203-433c-ba81-8f9f656ccc3f" />

</details>

<details>
<summary>Case 4 - non-US Region - changed specific default SE - PASSED</summary>

1. Continue from above test case
2. Navigate to brave://setting/search
3. Change default SE to something other than region's default SE

Example | Example
--- | ---
<img width="562" height="556" alt="Image" src="https://github.com/user-attachments/assets/a2cdd357-c628-451b-8d76-5088a6cbba25" /> | <img width="1193" height="355" alt="Image" src="https://github.com/user-attachments/assets/72277dcb-f601-4bc7-a410-9f165b8ee3e3" />

4. Navigate to brave://settings/clearBrowserData
5. Confirm no messaging is displayed for "Your search engine is..." on either "Delete browsing data" or "Delete data on exit" tabs

Example | Example
--- | ---
<img width="561" height="733" alt="Image" src="https://github.com/user-attachments/assets/ce103fe1-b80b-43af-b5c6-c7eca8e4b912" /> | <img width="557" height="598" alt="Image" src="https://github.com/user-attachments/assets/5738c26b-528d-4f9c-bcb0-064c3f09a415" />

</details>
